symengine/dict.cpp

```diff
@@ -204,6 +210,21 @@ bool umap_basic_basic_eq(const umap_basic_basic &a,
     return true;
 }
 
+#if defined(HAVE_SYMENGINE_PIRANHA)
+bool hash_set_eq(const hash_set &a,
+        const hash_set &b)
+{
+    // Can't be equal if # of elements in set differ:
+    if (a.size() != b.size()) return false;
+    // Loop over elements in "a":
+    for (auto &p: a) {
+        auto f = b.find(p);
+        if (f == b.end()) return false; // no such element in "b"
+    }
+    return true;
+}
+#endif
```

Comment:
This is ok for now, but I think we should write templated functions that work for any type, and then just call them for the given type, just like we do it already for printing. That way the actual logic (that can go wrong) is isolated to just a few places.